### DISCUSSION TITLE
Enhancement: Option to Toggle Collapse Right-Hand Sidebar

### DISCUSSION THREAD

**Discussion by richhayler** [craftcms/cms#8684] [Ideas]
I would love an option to be able to toggle left and right sidebars (as needed) to maximise the available content editing area to accomodate different device resolutions. While the left sidebar collapse behaviour is pretty good, the right-hand sidebar doesn’t have the option to collapse on demand. 

As an example, editing content on a 12” iPad Pro in portrait looks like this:

![IMG_0161](https://user-images.githubusercontent.com/2505643/58071737-e8cab400-7be0-11e9-8aa6-b2683c555ba2.png)
or
![ccc](https://user-images.githubusercontent.com/2505643/58071871-5080ff00-7be1-11e9-98d6-ea6eac94eed4.png)

It would be great if there was an option to toggle the right-hand sidebar so the main  content section could expand to take up all the available horizontal space to make for an improved authoring experience.

**Comment by richhayler** [comment]
Is there any chance this one could be considered for 3.2?

**Comment by brandonkelly** [comment]
Not likely for 3.2, but we have some UI enhancements slated for 3.3 so we should be able to get to this as part of that.

I’m positive I’ve seen Craft installs that have a collapse button just like in your screenshot, but I’m having a hard time finding it. :-/

**Comment by richhayler** [comment]
I’ve also seen something similar (it might have been a plugin to collapse the left nav). I’ll do some digging and maybe try to hack something together to replicate what I’m after.

**Comment by richhayler** [comment]
Hi @brandonkelly, Ok I've had a crack and made my first plugin :)

It could have easily been accomplished using the Control Panel CSS and JS plugins but I thought it would be a good opportunity to learn something new. 

The plugin can be installed via **composer require richhayler/stretch** 

Please feel free to use the CSS and JS if you would like to incorporate this into core in some way. The menu icon and styling could be improved of course but for now it does the job I need and makes it a tiny bit better editing experience for me on my iPad Pro.

<img width="1000" alt="Screenshot 2019-06-07 23 33 04" src="https://user-images.githubusercontent.com/2505643/59107759-a8a75780-897c-11e9-8816-42524a2d5fb5.png">

<img width="1000" alt="Screenshot 2019-06-07 23 33 17" src="https://user-images.githubusercontent.com/2505643/59107760-a8a75780-897c-11e9-80f4-d694ca26229a.png">
